### PR TITLE
Logging improvements: log context calling Optimize and IR size

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/Optimize.scala [hunk 1]
```diff
@@ -3,9 +3,10 @@ package is.hail.expr.ir
 import is.hail.utils._
 
 object Optimize {
-  private def optimize(ir0: BaseIR, noisy: Boolean, canGenerateLiterals: Boolean): BaseIR = {
+  private def optimize(ir0: BaseIR, noisy: Boolean, canGenerateLiterals: Boolean, context: Option[String]): BaseIR = {
+    val contextStr = context.map(s => s" ($s)").getOrElse("")
     if (noisy)
-      log.info("optimize: before:\n" + Pretty(ir0, elideLiterals = true))
+      log.info(s"optimize$context: before: IR size ${ IRSize(ir0) }: \n" + Pretty(ir0, elideLiterals = true))
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/Optimize.scala]: Do you want a space before context?

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/Optimize.scala]: no, the `Some` case adds a space.

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/expr/ir/Optimize.scala]: Oh, so `context` should be `contextStr`.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/Optimize.scala]: d'oh. good catch!

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/Optimize.scala]: done

**Pull request merged**